pkg/record/record.go

```diff
@@ -147,6 +148,19 @@ func (r Record) mapData(d Data) interface{} {
 	return nil
 }
 
+func (r Record) Clone() (Record, error) {
```

Comment:
What needs to be in `Record` for this method to return an error? 🤔

Reply:
The copier will return an error if there's something wrong with the source and destination of copying (source is a zero value, destination not addressable). IIRC, I got a panic with a struct with an unexported field in it. Now that I think of it, that could be a problem if we ever add unexported fields to `Record`.

Reply:
The source struct can't be zero since this method is declared on a value, and we are in charge of creating the destination value so we can guarantee it is addressable.

I tested copying a struct with an [unexported field](https://go.dev/play/p/y4aqr8SAam0) and it didn't error out, but I found some other strange behavior - the unexported field was _changed_ 🤨 This makes me a bit uneasy.

Anyway, my point is, if we are in charge of all the conditions that could make this fail, maybe we should just catch the error early and panic rather than return the error, since the caller can do nothing about it. Otherwise I challenge you to write a test that makes this function return an error 😛

Reply:
In the test for clone I changed `Name` to be unexported ([here](https://github.com/ConduitIO/conduit/pull/849/files#diff-7ac1828e3d69c68456d56641b56ffda6e981673226b9ea12cf895277ca6f582fR26)) and it failed... But now I see it's because of go-cmp.:D 



> but I found some other strange behavior - the unexported field was _changed_ raised_eyebrow This makes me feel a bit uneasy.

Not sure if you noticed, but the first argument in `copier.CopyWithOption` is the destination. You may have expected a source there.

> Anyway, my point is, if we are in charge of all the conditions that could make this fail, maybe we should just catch the error early and panic rather than return the error, since the caller can do nothing about it.

I actually had that in mind too. Panicking felt a bit too brutal, but I can definitely implement it that way.:)

Reply:
Oh my bad, I didn't RTFM 🤦‍♂️ It works like a charm, thanks for pointing that out. I see that even unexported fields are clones.

Well yeah, panicking feels brutal, but I think that if we can guarantee that the contents of `Record` don't make a difference and the error could only happen if we are using the library wrong, then I'd rather panic and write this down as an error that we intend to catch in tests. It makes the usage simpler.

Reply:
You're 200% right about the usage!

Reply:
Implemented as discussed above. Thanks for the suggestion @lovromazgon!